### PR TITLE
Introduced instanced rendering

### PR DESCRIPTION
**Pull request by guassmith**
Tiles, entities and items use instanced rendering now, entities and items may need their own shader, hard entities also need become instanced.

### CODE REVIEW

File: src/main/java/entity/Entity.java [hunk 1]
```diff
@@ -52,6 +53,10 @@ public boolean isVisible() {
 		return visible;
 	}
 
+	public Vec2f getXY() {
```
Comment (Bramelvix) [inline src/main/java/entity/Entity.java:56]: getLocation maybe?

File: src/main/java/main/Game.java [hunk 2]
```diff
@@ -139,16 +139,16 @@ private void init() throws Exception {
...
-		ByteBuffer temp = texture.buffer;
-		image.set(texture.width, texture.height, temp);
+		image.set(textureInfo.width, textureInfo.height, buffer);
 		GLFWImage.Buffer images = GLFWImage.malloc(1);
 		images.put(0, image);
 		glfwSetWindowIcon(window, images);
+
 		images.free();
 		image.free();
-
```
Comment (Bramelvix) [inline src/main/java/main/Game.java:151]: This changes nothing?

Comment (guassmith) [reply src/main/java/main/Game.java:151]: the buffer was called temp but it wasn't temp

File: src/main/java/entity/Entity.java (hunk 1 above)
Comment (guassmith) [reply src/main/java/entity/Entity.java:56]: if you want but that will take up a lot of space

File: src/main/java/map/Tile.java [hunk 3]
```diff
@@ -26,15 +29,24 @@
 	}
 
 	// render a tile
-	 void render(Vec2f pos, Vec2f offset) {
+	void render(Vec2f pos, Vec2f offset, ArrayList<Vec2f> tileRenderList, ArrayList<Vec2f> entityRenderList, ArrayList<Vec2f> itemRenderList) {
 		if (entity == null || entity.isTransparent() || !entity.isVisible()) {
-			sprite.draw(pos, offset);
+			//sprite.draw(pos, offset);
+			tileRenderList.add(sprite.getTexCoords());
+			//tileRenderList.add(sprite.getTexCoords().y);
```
Comment (Bramelvix) [inline src/main/java/map/Tile.java:36]: Why are these commented out? I Imagine they can be removed?

File: src/main/java/map/Tile.java [hunk 4]
```diff
@@ -26,15 +29,24 @@
...
+		} else {
+			tileRenderList.add(new Vec2f(0f,0f));
 		}
 		if (entity != null && !solid) {
-			entity.render(offset);
+			//entity.render(offset);
```
Comment (Bramelvix) [inline src/main/java/map/Tile.java:41]: Same here

File: src/main/java/main/Game.java (hunk 2 above)
Comment (Bramelvix) [reply src/main/java/main/Game.java:151]: Fair enough but you change way more than that

**Pull request merged**